pkg/kapis/monitoring/v1alpha3/handler.go

```diff
@@ -392,57 +396,89 @@ func (h handler) handleGrafanaDashboardImport(req *restful.Request, resp *restfu
 		grafanaDashboardContent = []byte(c)
 	}
 
+	isClusterCrd := namespace == ""
 	c := converter.NewConverter()
-	convertedDashboard, err := c.ConvertToDashboard(grafanaDashboardContent, true, "", grafanaDashboardName)
+	convertedDashboard, err := c.ConvertToDashboard(grafanaDashboardContent, isClusterCrd, namespace, grafanaDashboardName)
 	if err != nil {
 		api.HandleBadRequest(resp, nil, err)
 		return
 	}
 
+	ctx := context.TODO()
 	annotation := map[string]string{"kubesphere.io/description": entity.Description}
 
-	dashboard := monitoringdashboardv1alpha2.ClusterDashboard{
-		TypeMeta: v1.TypeMeta{
-			APIVersion: convertedDashboard.APIVersion,
-			Kind:       convertedDashboard.Kind,
-		},
-		ObjectMeta: v1.ObjectMeta{
-			Name:        convertedDashboard.Metadata["name"],
-			Annotations: annotation,
-		},
-		Spec: *convertedDashboard.Spec,
-	}
+	// a cluster scope dashboard or a namespaced dashboard with the same name cannot post.
+	if isClusterCrd {
+		clusterdashboard := monitoringdashboardv1alpha2.ClusterDashboard{
+			TypeMeta: v1.TypeMeta{
+				APIVersion: convertedDashboard.APIVersion,
+				Kind:       convertedDashboard.Kind,
+			},
+			ObjectMeta: v1.ObjectMeta{
+				Name:        convertedDashboard.Metadata["name"],
+				Annotations: annotation,
+			},
+			Spec: *convertedDashboard.Spec,
+		}
 
-	jsonDashbaord, err := json.Marshal(dashboard)
-	if err != nil {
-		api.HandleBadRequest(resp, nil, err)
-		return
-	}
+		objKey := runtimeclient.ObjectKey{
+			Namespace: "",
+			Name:      clusterdashboard.Name,
+		}
 
-	// a dashboard with the same name cannot post.
-	ctx := context.TODO()
-	_, err = h.k.Discovery().RESTClient().
-		Get().
-		AbsPath("/apis/monitoring.kubesphere.io/v1alpha2/clusterdashboards/" + dashboard.Name).
-		DoRaw(ctx)
+		err = h.rtClient.Get(ctx, objKey, &clusterdashboard)
 
-	if err == nil {
-		api.HandleBadRequest(resp, nil, errors.New("a dashboard with the same name already exists."))
-		return
-	}
+		if err == nil {
+			api.HandleBadRequest(resp, nil, errors.New("dashboards with the same name already exists."))
+			return
+		}
 
-	// create this dashboard
-	_, err = h.k.Discovery().RESTClient().
-		Post().
-		AbsPath("/apis/monitoring.kubesphere.io/v1alpha2/clusterdashboards").
-		Body(jsonDashbaord).
-		DoRaw(ctx)
+		// create this dashboard
+		err = h.rtClient.Create(ctx, &clusterdashboard)
```

Comment:
If err is not empty here, it may return directly rather than perform the subsequent creation.

Reply:
See below it already handled the not nil case. 
```
err = h.rtClient.Get(ctx, objKey, &clusterdashboard)
// can get it will be went failed.
if err == nil {
	api.HandleBadRequest(resp, nil, errors.New("dashboards with the same name already exists."))
	return
}

err = h.rtClient.Create(ctx, &clusterdashboard)
// returns with creation failed reason if cannot create
if err != nil {
	api.HandleBadRequest(resp, nil, err)
	return
}
// if err == nil , can create it, returns to the client the completed dashboard.
resp.WriteAsJson(clusterdashboard)
```

Reply:
The err below should be the creation error. Of course, ignoring the get error causes no problem for now.

Reply:
Yeah. The design is what the codes described.  If we return the get error, the creation process will not take an effect.